src/extension/background-script/connectors/connector.interface.ts

```diff
@@ -26,6 +26,26 @@ export interface ConnectorInvoice {
   type: "received";
 }
 
+// @Todo: extend missing properties accordingly
+export interface ConnectorTransaction {
```

Comment:
Maybe as many properties as possible should be shared with ConnectorInvoice (they could be extracted to a separate interface that both inherit from).

That way we don't have inconsistencies with the naming (e.g. preimage, amount, etc).

Reply:
Hmm, not sure if this makes any sense. We don't have a choice in the namings, they are dictated by the connector API results.

Reply:
This is the common connector interface, right? the individual connectors should map their results to this interface.

Reply:
I also understood it the way Roland does: All of the `getTransactions` implementations would return an array of `ConnectorTransaction` that maps any fields that differ in the connectors to this one common interface. 

Both `ConnectorTransaction` and `ConnectorInvoice` could implement a `ITransaction` interface to enforce the same naming of fields that are available on both types.

Reply:
> Hmm, not sure if this makes any sense. We don't have a choice in the namings, they are dictated by the connector API results.

I forgot, the mapping happens here yes...

https://github.com/lujakob/lightning-browser-extension/blob/aab05df5550c57b9c771dbe52ba0abab1dd41297/src/extension/background-script/connectors/lndhub.ts#L152-L163

Reply:
If possible it would be good to add more properties there - amount, date, id, custom_records. An (incoming) invoice and transaction should have many similar properties. I guess the fee only applies to transactions, and the type should be a different constant value ("received" for invoices and "sent" for transactions), everything else could be moved to the `ITransaction` interface?

Reply:
```
export interface ConnectorInvoice extends ITransaction {
  custom_records?: {
    "696969": string;
    "7629169": string;
    "5482373484": string;
  };
  id: string;
  settled: boolean;
  settleDate: number;
  totalAmount: string;
  type: "received";
}

// @Todo: extend missing properties accordingly
export interface ConnectorTransaction extends ITransaction {
  // @Todo: can this be the same custom type as in ConnectorInvoice above?
  custom_records?: Record<number, number>;
  fee: number;
  keysend: boolean;
  timestamp: number;
  type: string;
  value: number;
}

* for `custom_records` which type of the two should I use?
* `ConnectorTransaction` has no `id`
* can I make `ConnectorInvoice settleDate` and `ConnectorTransaction timestamp` into a shared `date`?

That's all the similarities I see..
 

```

Reply:
Maybe ```
custom_records?: {
    "696969": string;
    "7629169": string;
    "5482373484": string;
  } & Record<string, unknown>;
``` could be used. It should cover all cases.

I am not sure regarding the timestamp (whether it is the settled date or the created date?). It might be good to have both dates. For invoices, only settled invoices will show in the extension. But for transactions, we might want to show unsettled/pending transactions. So I think `settled` and `settledDate` could be moved to ITransaction.